lib/modules/datasource/docker/index.ts

```diff
@@ -453,6 +453,89 @@ export class DockerDatasource extends Datasource {
     }
   }
 
+  @cache({
+    namespace: 'datasource-docker-imageconfig',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      configDigest: string
+    ) => `${registryHost}:${dockerRepository}@${configDigest}`,
+    ttlMinutes: 1440,
+  })
+  public async getImageConfig(
+    registryHost: string,
+    dockerRepository: string,
+    configDigest: string
+  ): Promise<HttpResponse | null> {
+    logger.debug(
+      `getImageConfig(${registryHost}, ${dockerRepository}, ${configDigest})`
+    );
+    try {
+      const headers = await getAuthHeaders(
+        this.http,
+        registryHost,
+        dockerRepository
+      );
+      // istanbul ignore if: Should never be happen
+      if (!headers) {
+        logger.debug('No docker auth found - returning');
+        return null;
+      }
+      const url = `${registryHost}/v2/${dockerRepository}/blobs/${configDigest}`;
+      return await this.http.get(url, {
+        headers,
+        noAuth: true,
+      });
+    } catch (err) /* istanbul ignore next */ {
```

Comment:
we should return undefined for any errors beside 404. The cache decorator will discard `undefined`  return values.
On same renovate run, the memory cache should safe us from multiple http requests, so we can probably simply rethrow any error beside 404

Reply:
I still think we should simply pass errors up. They will be catched on caller anyways and makes testing / maintenance  a lot easier.

@rarkins WDYT?

Reply:
Fine by me

Reply:
@viceice, I've now adapted this in [b5ec947](https://github.com/renovatebot/renovate/pull/16554/commits/b5ec9470993a04a5b13bc388aeb99a2dc4a19fcb) and basically aligned with how `getDigest()` does error handling.
I hope that's what you meant, otherwise please let me know.

Reply:
no, remove error handling here completly, all errors should be passed up